Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs

```diff
@@ -0,0 +1,239 @@
+        private WindowsMicrophoneStream micStream = null;
+
+        /// <summary>
+        /// The average amplitude of the sound captured during the most recent microphone update.
+        /// </summary>
+        private float averageAmplitude = 0.0f;
+
+#if UNITY_EDITOR
+        /// <summary>
+        /// Cached material values used to restore initial settings when running the demo in the editor.
+        /// </summary>
+        private Color defaultMaterialColor = Color.black;
+        private int defaultWireThickness = 0;
+#endif // UNITY_EDITOR
+
+        private void Awake()
+        {
+            // We do not wish to play the ambient room sound from the audio source.
+            gameObject.GetComponent<AudioSource>().volume = 0.0f;
+
+            if (VisibleMaterial != null)
+            {
+#if UNITY_EDITOR
+                // Cache the initial material settings.
+                defaultMaterialColor = VisibleMaterial.GetColor("_WireColor");
+                defaultWireThickness = VisibleMaterial.GetInt("_WireThickness");
+#endif // UNITY_EDITOR
+
+                VisibleMaterial.SetColor("_WireColor", Color.blue);
+            }
+
+            micStream = new WindowsMicrophoneStream();
+            if (micStream == null)
+            {
+                Debug.Log("Failed to create the Windows Microphone Stream object");
+            }
+
+            micStream.Gain = inputGain;
+
+            // Initialize the microphone stream.
+            WindowsMicrophoneStreamErrorCode result = micStream.Initialize(WindowsMicrophoneStreamType.HighQualityVoice);
+            if (result != WindowsMicrophoneStreamErrorCode.Success)
+            {
+                Debug.Log($"Failed to initialize the microphone stream. {result}");
+                return;
+            }
+
+            // Start the microphone stream.
+            // Do not keep the data and do not preview.
+            result = micStream.StartStream(false, false);
+            if (result != WindowsMicrophoneStreamErrorCode.Success)
+            {
+                Debug.Log($"Failed to start the microphone stream. {result}");
+            }
+        }
+
+        private void OnDestroy()
+        {
+            if (micStream == null) { return; }
+
+            // Stop the microphone stream.
+            WindowsMicrophoneStreamErrorCode result = micStream.StopStream();
+            if (result != WindowsMicrophoneStreamErrorCode.Success)
+            {
+                Debug.Log($"Failed to stop the microphone stream. {result}");
+            }
+
+            // Uninitialize the microphone stream.
+            micStream.Uninitialize();
+            micStream = null;
+
+#if UNITY_EDITOR
+            // Restore the initial material settings.
+            if (VisibleMaterial != null)
+            {
+                VisibleMaterial.SetColor("_WireColor", defaultMaterialColor);
+                VisibleMaterial.SetInt("_WireThickness", defaultWireThickness);
+            }
+#endif // UNITY_EDITOR
```

Comment:
Is this editor-only to ensure we don't accidentally save changes to the material file directly? Can we instead guard against that however we're accessing / editing the material?

Reply:
correct. i will look into a non-editor way to do this

Reply:
i removed the editor guards. the demo now restores the material values on destroy on device as well as in editor